File: src/java/org/orbeon/oxf/xforms/control/controls/RepeatIterationControl.java
/**
 *  Copyright (C) 2006 Orbeon, Inc.
 *
 *  This program is free software; you can redistribute it and/or modify it under the terms of the
 *  GNU Lesser General Public License as published by the Free Software Foundation; either version
 *  2.1 of the License, or (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *  See the GNU Lesser General Public License for more details.
 *
 *  The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
 */
package org.orbeon.oxf.xforms.control.controls;

import org.orbeon.oxf.xforms.XFormsContainingDocument;
import org.orbeon.oxf.xforms.control.XFormsControl;
import org.orbeon.oxf.xforms.control.XFormsSingleNodeControl;
import org.orbeon.oxf.pipeline.api.PipelineContext;

/**
 * Represents xforms:repeat iteration information.
 *
 * This is not really a control, but an abstraction for xforms:repeat branches.
 */
public class RepeatIterationControl extends XFormsSingleNodeControl {
    private int iteration;
    public RepeatIterationControl(XFormsContainingDocument containingDocument, XFormsControl parent, int iteration) {
        super(containingDocument, parent, null, "xxforms-repeat-iteration", parent.getEffectiveId());
        this.iteration = iteration;
    }

    public int getIteration() {
        return iteration;
    }

    protected void evaluate(PipelineContext pipelineContext) {
        // Just get the MIPs, don't call super.evaluate(), since that would also try to get label, etc.
        getMIPsIfNeeded();
    }

    public boolean isStaticReadonly() {
        return false;
    }

    public String getType() {
        return null;
    }

    public boolean equals(Object obj) {
        if (!super.equals(obj))
            return false;
        final RepeatIterationControl other = (RepeatIterationControl) obj;
        return this.iteration == other.iteration;
    }
}

File: xforms/jvm/src/main/scala/org/orbeon/oxf/xforms/analysis/controls/RepeatIterationControl.scala
/**
 *  Copyright (C) 2007 Orbeon, Inc.
 *
 *  This program is free software; you can redistribute it and/or modify it under the terms of the
 *  GNU Lesser General Public License as published by the Free Software Foundation; either version
 *  2.1 of the License, or (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *  See the GNU Lesser General Public License for more details.
 *
 *  The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
 */
package org.orbeon.oxf.xforms.analysis.controls

import org.orbeon.dom.Element
import org.orbeon.oxf.xforms.analysis.{ElementAnalysis, PartAnalysisImpl}
import org.orbeon.oxf.xforms.event.XFormsEvents._
import org.orbeon.xforms.xbl.Scope

class RepeatIterationControl(
  part     : PartAnalysisImpl,
  index    : Int,
  element  : Element,
  parent   : Option[ElementAnalysis],
  preceding: Option[ElementAnalysis],
  scope    : Scope
) extends ContainerControl(part, index, element, parent, preceding, scope)
     with RequiredSingleNode {

  override protected def externalEventsDef = super.externalEventsDef + XXFORMS_REPEAT_ACTIVATE
  override val externalEvents              = externalEventsDef
}